### PR TITLE
Trigger release workflow manually

### LINKED ISSUE #3815: Optimize branch name and pull method
1. The branch name is the same as the tag name now, we should prefix the branch name with "release-". Doc and workflow, etc. also need to be modified.
2. The vX.Y version should be in a branch, and the bug fix version is identified by tag.

### PR DESCRIPTION
**Pull request by yixinglu**
## What type of PR is this?
- [ ] bug
- [ ] feature
- [X] enhancement

#### Issue(s) number: 

close #3815 

#### Description:

when releasing a nebula PATCH version(z version of x.y.z), create git tag on one same branch rather than a new branch, such as:

```text
release-3.2   -----+----------+------- branch
                    \          \
                    v3.2.0    v3.2.1   tag
```

## How do you solve it?

use github action manual trigger to package rc version.

## Checklist:
Tests:
- [ ] Unit test(positive and negative cases)
- [ ] Function test
- [ ] Performance test
- [X] N/A

Affects:
- [ ] Documentation affected (Please add the label if documentation needs to be modified.)
- [ ] Incompatibility (If it breaks the compatibility, please describe it and add the label.）
- [ ] If it's needed to cherry-pick (If cherry-pick to some branches is required, please label the destination version(s).)
- [ ] Performance impacted: Consumes more CPU/Memory

## Release notes:
None

Please confirm whether to be reflected in release notes and how to describe:
> ex. Fixed the bug .....
None

### CODE REVIEW

**Comment by Sophie-Xie** [comment]
Thanks!

And I have some questions, if I enter the version number, is it v3.0.2 or 3.0.2?  And Is the name of github tag still v3.0.2?

**Comment by yixinglu** [comment]
> Thanks!
> 
> And I have some questions, if I enter the version number, is it v3.0.2 or 3.0.2? And Is the name of github tag still v3.0.2?

Input version of the workflow is following format: `3.0.2`, there's an example in this parameter description. The tag name should be same as before, such as: `v3.0.2`.

File: .gitignore [hunk 1]
```diff
@@ -38,6 +38,7 @@ modules/
 
 # tests
 !tests/Makefile
+tests/secrets
```
Comment (Aiee) [inline .gitignore:41]: Ent version require this directory

File: .github/workflows/rc.yml [hunk 2]
```diff
@@ -1,9 +1,12 @@
...
 on:
-  push:
-    branches:
-      - 'v[0-9]+.*'
+  workflow_dispatch:
+    inputs:
+      version:
+        description: "version such as '3.1.0'"
+        required: true
+        default: '3.1.0'
```
Comment (jievince) [inline .github/workflows/rc.yml:9]: hardcode 3.1.0?

**Review by Aiee — DISMISSED** [review]

File: .github/workflows/rc.yml (hunk 2 above)
Comment (HarrisChu) [reply .github/workflows/rc.yml:9]: prefer to do not use default, if not provide version, it would be errors

Comment (yixinglu) [reply .github/workflows/rc.yml:9]: make sense!

File: .github/workflows/pull_request.yml [hunk 3]
```diff
@@ -5,7 +5,7 @@ on:
     types: [synchronize, reopened, labeled]
     branches:
       - master
-      - 'v[0-9]+.*'
+      - 'release-**'
```
Comment (HarrisChu) [inline .github/workflows/pull_request.yml:8]: typo?

should match release-3.1

**Review by HarrisChu — APPROVED** [review]

**Pull request merged**